criu/net.c

```diff
@@ -3081,6 +3082,14 @@ static inline int nftables_lock_network_internal(void)
 	if (!nft)
 		return -1;
 
+	fp = fdopen(log_get_fd(), "w");
+	if (!fp) {
+		pr_perror("fdopen() failed");
+		goto err3;
+	}
+	nft_ctx_set_output(nft, fp);
+	nft_ctx_set_error(nft, fp);
```

Comment:
The code in `nftables_lock_network_internal()` and `nftables_network_unlock()` looks very similar.

Would it make sense to create a function?
```C
static inline int nftables_set_logfd(struct nft_ctx *nft)
{
	FILE *fp;
	fp = fdopen(log_get_fd(), "w");
	if (!fp) {
		pr_perror("fdopen() failed");
		return -1;
	}
	nft_ctx_set_output(nft, fp);
	nft_ctx_set_error(nft, fp);

	fflush(fp);
	fclose(fp);
	return 0;
}
```
We can use this function in `nftables_lock_network_internal()`:
```C
if (nftables_set_logfd(nft)) {
	goto err2;
}
```
and in `nftables_network_unlock()`:
```C
if (nftables_set_logfd(nft)) {
	nft_ctx_free(nft);
	return -1;
}
```

Reply:
```
The nft_ctx_set_output() and nft_ctx_set_error() functions set the output or error file pointer in ctx to the value of fp. They return the previous value to aid in temporary file pointer overrides. On error, these functions return NULL. 
```

nftables_set_logfd can't close fp...